### PR TITLE
Adding CLI for the migration tool

### PR DESCRIPTION
**Pull request by rajansandeep**
Adds the CLI as defined [here](https://github.com/coredns/deployment/tree/master/kubernetes/migration#command-line-converter)

Adds support to basic commands such as `migrate`, `deprecated`, `default`,  `removed`, `unsupported` and `vaildversions`

### CODE REVIEW

File: kubernetes/corefile-tool/cmd/default.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	}
+	defaultCmd.Flags().String("k8sversion", "", "The Kuberenetes version for which you are checking the default.")
+	defaultCmd.Flags().String("corefile", "", "Required: The path where your Corefile is located.")
+	defaultCmd.MarkFlagRequired("corefile")
+
+	return defaultCmd
+}
+
+// defaultCorefileFromPath takes the path where the Corefile is located and checks
+// if the Corefile is the default for a that version of Kubernetes.
```
Comment (fturib) [inline kubernetes/corefile-tool/cmd/default.go]: typo with "for a"

File: kubernetes/corefile-tool/cmd/default.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+func defaultCorefileFromPath(k8sVersion, corefilePath string) (bool, error) {
+	if _, err := os.Stat(corefilePath); os.IsNotExist(err) {
```
Comment (fturib) [inline kubernetes/corefile-tool/cmd/default.go]: all the lines from #43 to #51 are the same for all function "FromPath: ...
Can you just build a small utility function "getCorefileFrompath(path string)(string error) ?

File: kubernetes/corefile-tool/cmd/root.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+)
+
+// CorefileTool represents the base command for the corefile-tool.
+func CorefileTool() *cobra.Command {
+	rootCmd := &cobra.Command{
+		Use:   "corefile-tool",
+		Short: "A brief description of your application",
+		Long: dedent.Dedent(`
+
+			    ┌──────────────────────────────────────────────────────────┐
```
Comment (fturib) [inline kubernetes/corefile-tool/cmd/root.go]: Good !

File: kubernetes/corefile-tool/cmd/default.go (hunk 2 above)
Comment (rajansandeep) [reply kubernetes/corefile-tool/cmd/default.go]: Done!

File: kubernetes/corefile-tool/README.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+## Usage
+
+Use the following syntax to run the `corefile-tool` command:
+
+`corefile-tool [command] [flags]`
+
+where `command`, `flags` are:
+
+- `command`: The operation you want to perform.
```
Comment (chrisohaver) [inline kubernetes/corefile-tool/README.md]: Need to define what the valid commands are.  I think these are in the "Operations" section, but the terminology  is not consistent command/operation, so it's confusing.

File: kubernetes/corefile-tool/README.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+## Usage
+
+Use the following syntax to run the `corefile-tool` command:
+
+`corefile-tool [command] [flags]`
+
+where `command`, `flags` are:
+
+- `command`: The operation you want to perform. 
+- `flags`  : Specifies flags required to carry out the operations.
```
Comment (chrisohaver) [inline kubernetes/corefile-tool/README.md]: Need to define what the valid flags are per command.

File: kubernetes/corefile-tool/README.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+corefile-tool unsupported --from 1.4.0 --to 1.5.0 --corefile /path/to/Corefile
+```
+
+```bash
+# See removed plugins CoreDNS from v1.4.0 to v1.5.0. 
+corefile-tool removed --from 1.4.0 --to 1.5.0 --corefile /path/to/Corefile
+```
+
+```bash
+# Migrate CoreDNS from v1.4.0 to v1.5.0 and handle deprecations .
```
Comment (chrisohaver) [inline kubernetes/corefile-tool/README.md]: Should explain what "handle deprecations" means.

**Comment by chrisohaver** [comment]
Can you also add a wrapper for [Released()](https://github.com/coredns/deployment/pull/154/files#diff-11d4c31d0972a2b605e5fbb2836f9970R236)?

**Pull request merged**